packages/world/src/systems/TransferSystem.sol

```diff
@@ -24,56 +23,46 @@ contract TransferSystem is System {
     EntityId caller,
     EntityId from,
     EntityId to,
-    SlotTransfer[] memory slotTransfers,
+    SlotTransfer[] memory transfers,
     bytes calldata extraData
   ) public {
     caller.activate();
 
-    if (caller != from) {
-      caller.requireConnected(from);
-    }
-    from.requireConnected(to);
-
-    transferEnergyToPool(caller, SMART_CHEST_ENERGY_COST);
-
-    (EntityId[] memory entities, ObjectAmount[] memory objectAmounts) =
-      InventoryUtils.getObjectsAndEntities(from, slotTransfers);
-
-    InventoryUtils.transfer(from, to, slotTransfers);
-
-    // Note: we call this after the transfer state has been updated, to prevent re-entrancy attacks
-    TransferLib._onTransfer(caller, from, to, entities, objectAmounts, extraData);
-  }
-}
+    EntityId target;
 
-library TransferLib {
-  function _onTransfer(
-    EntityId caller,
-    EntityId from,
-    EntityId to,
-    EntityId[] memory entities,
-    ObjectAmount[] memory objectAmounts,
-    bytes calldata extraData
-  ) public {
-    EntityId target = _getTarget(caller, from, to);
-
-    require(ObjectType._get(target) != ObjectTypes.Player, "Cannot transfer to player");
-
-    bytes memory onTransfer =
-      abi.encodeCall(ITransferHook.onTransfer, (caller, target, from, to, objectAmounts, entities, extraData));
+    // Transferring within the same inventory
+    if (from == to) {
+      if (caller != from) {
+        caller.requireConnected(from);
```

Comment:
Do we need a player check here as well? What if I call transfer with: caller = player A, from = player B, to = player B

Ie moving objects around in another players inventory

Reply:
ah lmao you are right

Reply:
would be good to add a test for this case as well

Reply:
yeah we need a test sprint for sure, rn I'm just trying to get everything to a working state